mono/mini/il2cpp-stubs.cpp

```diff
@@ -738,8 +736,8 @@ Il2CppMonoObject* il2cpp_mono_field_get_value_object_checked(Il2CppMonoDomain* d
 
 Il2CppMonoObject* il2cpp_mono_object_new_checked(Il2CppMonoDomain* domain, Il2CppMonoClass* klass, MonoError* error)
 {
-	IL2CPP_ASSERT(0 && "This method is not yet implemented");
-	return NULL;
+	mono_error_init(error);
+	return (Il2CppMonoObject*)il2cpp::vm::Object::New((Il2CppClass*)klass);
```

Comment:
We could call `il2cpp_object_new` here, although I'm on the fence about this one. Since we can't replace this stub entirely (due to the `MonoError` parameter).

Reply:
Per our previous discussion, we don't make il2cpp-api calls in the stubs file, we use the C++ api.